### PR TITLE
gdb-svd: add 'dump' command to store registers values to file

### PR DESCRIPTION
**Pull request by ABOSTM**
gdb-svd: add 'dump' command to store registers values to file

ex:
(gdb) svd dump c:\tmps\dump.txt
(gdb) svd dump c:\tmps\dump.txt ADC1

### CODE REVIEW

File: gdb-svd.py [hunk 1]
```diff
@@ -353,3 +361,45 @@ def invoke(self, arg, from_tty):
...
+                args = str(text).split(" ")
+                if len(args) > 2:
+                        return gdb.COMPLETE_NONE
+
+                return GdbSvdCmd.complete(self, text, word)
+
+        def invoke(self, arg, from_tty):
+                args = str(arg).split(" ")
+                if len(args) < 1 or len(args) > 2:
+                        gdb.write("Invalide parameter\n")
```
Comment (erwango) [inline gdb-svd.py]:
```suggestion
                        gdb.write("Invalid parameter\n")
```

**Comment by erwango** [comment]
Since the final target of this feature is to be able to compare 2 files and directly see the differences of configuration between 2 set ups, I think it would be great to take full advantage of the SVD file and dump the whole values with whole description ( the result you get with a `svd get <perph>` applied to all periph...
This would be a bit longer I imagine, but you only do it once and this is used for "off debugger" analysis. Then, when comparing 2 files you not only get where the difference is but also what it impacts and then you can focus directly on registers where difference makes sense in your use case.

**Comment by erwango** [comment]
> Since the final target of this feature is to be able to compare 2 files and directly see the differences of configuration between 2 set ups, I think it would be great to take full advantage of the SVD file and dump the whole values with whole description ( the result you get with a `svd get <perph>` applied to all periph... This would be a bit longer I imagine, but you only do it once and this is used for "off debugger" analysis. Then, when comparing 2 files you not only get where the difference is but also what it impacts and then you can focus directly on registers where difference makes sense in your use case.

Ok, this is indeed what you get when opening the file, this is great...
I've been tricked by the console output:
```
(gdb) svd dump wl_zephyr_pm.txt
Print to file: wl_zephyr_pm.txt
0x40012400: 00000000 

0x40012404: 00000000 

```

**Comment by erwango** [comment]
One request:
In dump mode, the syntax highlighting need to be disabled as this makes file comparison much less readable

**Comment by ABOSTM** [comment]
> One request: In dump mode, the syntax highlighting need to be disabled as this makes file comparison much less readable

Added :-)

**Comment by 1udo6arre** [comment]
Hi Alex

great feature, I just fixed an issue on command completion.
I merge the pull request, and I will push the fix after.

Regard
Ludo

**Pull request merged**